supervisor/addons/addon.py

```diff
@@ -170,6 +172,15 @@ def in_progress(self) -> bool:
 
     async def load(self) -> None:
         """Async initialize of object."""
+        if MAP_CONFIG in self.map_volumes:
```

Comment:
Let's add this to: https://github.com/home-assistant/supervisor/blob/9951ceda7b7b234e884d43f562a89ee9457b5e68/supervisor/addons/validate.py#L199

Reply:
I can add a warning here but I can't migrate it. If using `config` in this option then the home assistant config folder gets mounted to `/config`. Which means the addon's new public config folder cannot get mounted there and is therefore not mounted at all into the container. I can't just migrate one setting to the other or the addon will break as it will be expecting the Home Assistant folder at `/config`, not at `/homeassistant`